### PR TITLE
Docs: add 'stricter rule config validating' in migrating docs

### PR DESCRIPTION
**Pull request by aladdin-add**
**What is the purpose of this pull request? (put an "X" next to item)**

[x] Documentation update

**What changes did you make? (Give an overview)**
refs #9505

**Is there anything you'd like reviewers to focus on?**

wording improvement?

### CODE REVIEW

File: docs/user-guide/migrating-to-6.0.0.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ The lists below are ordered roughly by the number of users each change is expect
 1. [`eslint:recommended` has been updated](#eslint-recommended-changes)
 1. [Plugins and shareable configs are no longer affected by ESLint's location](#package-loading-simplification)
 1. [The default parser now validates options more strictly](#espree-validation)
+1. [Rule configuration are vlidated more strictly](#rule-config-validating)
```
Comment (kaicataldo) [inline docs/user-guide/migrating-to-6.0.0.md]: Small typo: `vlidated` -> `validated`

File: docs/user-guide/migrating-to-6.0.0.md [hunk 2]
```diff
@@ -111,6 +112,22 @@ If you use a config file located outside of a local project (with the `--config`
 
 **Related issue(s):** [eslint/eslint#9687](https://github.com/eslint/eslint/issues/9687), [eslint/espree#384](https://github.com/eslint/espree/issues/384)
 
+## <a name="rule-config-validating"></a> Rule configuration are vlidated more strictly
```
Comment (kaicataldo) [inline docs/user-guide/migrating-to-6.0.0.md]: Small typo: `vlidated` -> `validated`

File: docs/user-guide/migrating-to-6.0.0.md [hunk 3]
```diff
@@ -111,6 +112,22 @@ If you use a config file located outside of a local project (with the `--config`
...
+
+To catch config errors eariler, ESLint v6 will report a linting error if you are trying to configure a non-existent rule.
```
Comment (kaicataldo) [inline docs/user-guide/migrating-to-6.0.0.md]: Small typo: `eariler` -> `earlier`

**Comment by kaicataldo** [comment]
A few comments, but otherwise LGTM!

File: docs/user-guide/migrating-to-6.0.0.md [hunk 4]
```diff
@@ -111,6 +112,22 @@ If you use a config file located outside of a local project (with the `--config`
 
 **Related issue(s):** [eslint/eslint#9687](https://github.com/eslint/eslint/issues/9687), [eslint/espree#384](https://github.com/eslint/espree/issues/384)
 
+## <a name="rule-config-validating"></a> Rule configuration are validated more strictly
+
+To catch config errors earlier, ESLint v6 will report a linting error if you are trying to configure a non-existent rule.
```
Comment (g-plane) [inline docs/user-guide/migrating-to-6.0.0.md]:
```suggestion
To catch config errors earlier, ESLint v6 will report a linting error if you are trying to configuring a non-existent rule.
```

Comment (platinumazure) [reply docs/user-guide/migrating-to-6.0.0.md]: I think the original wording ("to configure") is correct here.

Comment (g-plane) [reply docs/user-guide/migrating-to-6.0.0.md]: Does the semantic here mean "attempt to do something"?

Comment (aladdin-add) [reply docs/user-guide/migrating-to-6.0.0.md]: yes. try to do sth.

Comment (kaicataldo) [reply docs/user-guide/migrating-to-6.0.0.md]: That’s right! This can be rewritten as “...if you are attempting to configure a non-existent rule.”

Comment (g-plane) [reply docs/user-guide/migrating-to-6.0.0.md]: So this should be `try doing sth` or `attempt to do sth`, not `try to do sth`.

Comment (kaicataldo) [reply docs/user-guide/migrating-to-6.0.0.md]: I think it’s fine as it is. “Try doing” would be grammatically incorrect in this case.

Comment (g-plane) [reply docs/user-guide/migrating-to-6.0.0.md]: OK.

**Pull request merged**